web/client/components/map/enhancers/__tests__/overrideMapProps-test.js

```diff
@@ -0,0 +1,68 @@
+
+/*
+ * Copyright 2021, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import expect from 'expect';
+import withScalesDenominators from '../withScalesDenominators';
+import mockData from '../../../../test-resources/testConfig.json';
+import PropTypes from 'prop-types';
+
+class MockApp extends React.Component {
+
+    static propTypes = {
+        zoom: PropTypes.number
+    };
+
+    constructor(props) {
+        super(props);
+    }
+
+    render() {
+        return (
+            <div className="MockApp">
+                <p>{this.props.zoom}</p>
+            </div>
+        );
+    }
+}
+
+export default MockApp;
+
+let MockComp = withScalesDenominators(MockApp);
+
+describe('Test get resolutions from scales', () => {
+    beforeEach((done) => {
+        document.body.innerHTML = '<div id="container"></div>';
+        setTimeout(done);
+    });
+    afterEach((done) => {
+        ReactDOM.unmountComponentAtNode(document.getElementById("container"));
+        document.body.innerHTML = '';
+        setTimeout(done);
+    });
+
+    it('Test componet is rendered', () => {
+        ReactDOM.render(<MockComp />, document.getElementById("container"));
+        const container = document.getElementById('container');
+        const el = container.querySelector('.MockApp');
+        expect(el).toExist();
+    });
+
+    it('Test component props', () => {
+        ReactDOM.render(<MockComp {...mockData.map} />, document.getElementById("container"));
+        const container = document.getElementById('container');
+        const el = container.querySelector('.MockApp');
+        expect(el).toExist();
+        const text = document.getElementsByClassName('MockApp')[0].childNodes[0].innerHTML;
+        expect(text).toBe('' + mockData.map.zoom);
```

Comment:
If possible it could be useful to show with a test the `map.view.scales` prop transformation to `map.mapOptions.view.resolutions` instead of the zoom prop:

`<MockComp map={{ view: { scales: [ 50000, 25000, 10000 ] } }} />`

```js
function MockApp({ map }) {
  return (
    <ul>
      {map.mapOptions.view.resolutions.map(res => <li key={res}>{res}</li>)}
    </ul>
  );
}

```